src/vm/compiler.c

```diff
@@ -1037,6 +1037,28 @@ int parseEscapeSequences(char *string, int length) {
                     string[i + 1] = '\v';
                     break;
                 }
+                case 'x': {
+                    if (string[i + 2] == '\0' || string[i + 3] == '\0') {
+                        error(parser, "\\x escape code expects format \\xhh");
+                        break;
+                    }
+
```

Comment:
Can we add an additional check here to make sure the given characters are in fact valid hex (0123456789abcdefABCDEF)?

Reply:
That should happen below in the strtol check, but potentially an easier check is checking for the char being in that range

Reply:
Oh my bad, I just noticed that now. I'll defer to you. :)

Reply:
No worries! Thanks for double checking anyways